src/solana/solanaEngine.js

```diff
@@ -0,0 +1,390 @@
+// @flow
+
+import * as solanaWeb3 from '@solana/web3.js'
+import { bns } from 'biggystring'
+import { asNumber } from 'cleaners'
+import {
+  type EdgeFetchFunction,
+  type EdgeSpendInfo,
+  type EdgeTransaction,
+  type EdgeWalletInfo,
+  type JsonObject,
+  InsufficientFundsError
+} from 'edge-core-js/types'
+
+import { CurrencyEngine } from '../common/engine.js'
+import { asyncWaterfall, cleanTxLogs, getOtherParams } from '../common/utils.js'
+import { SolanaPlugin } from './solanaPlugin.js'
+import {
+  type RpcGetTransaction,
+  type RpcSignatureForAddress,
+  type SolanaOtherData,
+  type SolanaSettings,
+  asRecentBlockHash,
+  asRpcBalance,
+  asRpcGetTransaction
+} from './solanaTypes.js'
+
+const { PublicKey, Keypair, SystemProgram, Transaction } = solanaWeb3
+
+const ACCOUNT_POLL_MILLISECONDS = 5000
+const BLOCKCHAIN_POLL_MILLISECONDS = 20000
+const TRANSACTION_POLL_MILLISECONDS = 3000
+
+export class SolanaEngine extends CurrencyEngine {
+  keypair: Keypair
+  feePerSignature: string
+  recentBlockhash: string
+  chainCode: string
+  otherData: SolanaOtherData
+  fetchCors: EdgeFetchFunction
+  settings: SolanaSettings
+
+  constructor(
+    currencyPlugin: SolanaPlugin,
+    walletInfo: EdgeWalletInfo,
+    opts: any, // EdgeCurrencyEngineOptions
+    fetchCors: EdgeFetchFunction
+  ) {
+    super(currencyPlugin, walletInfo, opts)
+    this.keypair = currencyPlugin.createKeyPair(
```

Comment:
This should not happen on every engine creation. createKeyPair uses slow cryptography. This should only be done when needed to sign a transaction. The public key should be created at the time you import or create the private key and be returned in the keys object.